modules/tinymce/src/core/test/ts/browser/delete/DivDeleteTest.ts

```diff
@@ -0,0 +1,31 @@
+import { before, describe, it } from '@ephox/bedrock-client';
+import { PlatformDetection } from '@ephox/sand';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.delete.DivDeleteTest', () => {
+  const isFirefox = PlatformDetection.detect().browser.isFirefox();
+
+  const hook = TinyHooks.bddSetup<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [], true);
+
+  before(function () {
+    // Firefox doesn't unwrap DIV elements if you delete inside them and it's the only child element in the BODY
```

Comment:
Don't we want our behaviour to be as similar as possible across browsers?

Reply:
Yes at some point I would want to create a custom delete not a bunch of patches on the native delete. Not sure this is the time to override this as well. With the method to are applying now it's hard to know if you catched all the scenarios. It will be a bigger task to rewrite this to custom code.

Reply:
I guess maybe we could first detect what the block is then execute native delete then wrap it back in that block if it was deleted. That would solve this particular case.

Reply:
Yeah that approach seems to work but yes another patch over the delete. 😄